### PR TITLE
DOC: Add random generator exponential example

### PR DESCRIPTION
**Pull request by lzha97**
This PR adds an example for the random generator exponential function

Partially addresses Issue #22270 as part of GH22 OSX day

### CODE REVIEW

File: numpy/random/_generator.pyx [hunk 1]
```diff
@@ -389,6 +389,23 @@ cdef class Generator:
...
+        >>> n = 10000
+        >>> z = np.random.default_rng().exponential(scale=4, size=n)
+
+        What is the probability an agent will spend between 4
+        to 5 minutes with a customer? 
+
+        >>> x = ((d < 5).sum())/n 
+        >>> y = ((d < 4).sum())/n
+        >>> x-y
+        0.08 # may vary
```
Comment (HaoZeke) [inline numpy/random/_generator.pyx:395]:
```suggestion
        >>> x-y # will vary
```

File: numpy/random/mtrand.pyx [hunk 2]
```diff
@@ -533,6 +533,20 @@ cdef class RandomState:
...
+        agents and each agent spends on average 4 minutes with a 
+        customer.
+        >>> n = 10000
+        >>> z = np.random.exponential(scale=4, size=n)
+        What is the probability an agent will spend between 4
+        to 5 minutes with a customer? 
+        >>> x = ((d < 5).sum())/n 
+        >>> y = ((d < 4).sum())/n
+        >>> x-y
+        0.08 # may vary
```
Comment (HaoZeke) [inline numpy/random/mtrand.pyx:548]:
```suggestion
        >>> x-y # will vary
```

**Review by HaoZeke — COMMENTED** [review]
Neat, minor nit aside :)

File: numpy/random/_generator.pyx (hunk 1 above)
Comment (lzha97) [reply numpy/random/_generator.pyx]: Thank you! I've made the edit

**Comment by lzha97** [comment]
Does anyone know why the formatting is so off in CircleCI? 
image

**Comment by lzha97** [comment]
I was using VScode like this
image

File: numpy/random/_generator.pyx (hunk 1 above)
Comment (bsipocz) [reply numpy/random/_generator.pyx]: yes, sorry, I think this was my mistake on the call.

**Comment by WarrenWeckesser** [comment]
@lzha97, there must be a blank line at the beginning and end of each block of interactive shell prompts, e.g.
```
        A real world example: Assume a company has 10000 customer support 
        agents and each agent spends on average 4 minutes with a 
        customer.

        >>> n = 10000
        >>> z = np.random.default_rng().exponential(scale=4, size=n)

        What is the probability an agent will spend between 4
```

File: numpy/random/_generator.pyx [hunk 3]
```diff
@@ -380,6 +380,20 @@ cdef class Generator:
         out : ndarray or scalar
             Drawn samples from the parameterized exponential distribution.
 
+        Examples
+        --------
+        A real world example: Assume a company has 10000 customer support 
+        agents and each agent spends on average 4 minutes with a 
+        customer.
+        >>> n = 10000
+        >>> z = np.random.default_rng().exponential(scale=4, size=n)
```
Comment (rossbar) [inline numpy/random/_generator.pyx:389]: > Does anyone know why the formatting is so off in CircleCI?

There must be a blank line between the explanatory text and the actual executable example in order for the docstring to parse correctly:

```suggestion
        A real world example: Assume a company has 10000 customer support 
        agents and each agent spends on average 4 minutes with a 
        customer.
        
        >>> n = 10000
        >>> z = np.random.default_rng().exponential(scale=4, size=n)
```

Comment (lzha97) [reply numpy/random/_generator.pyx:389]: Ahh my bad, thank you!

Comment (lzha97) [reply numpy/random/_generator.pyx]: That fixed it, thank you!
image

image

File: numpy/random/_generator.pyx [hunk 4]
```diff
@@ -380,6 +380,23 @@ cdef class Generator:
...
             Drawn samples from the parameterized exponential distribution.
 
+        Examples
+        --------
+        A real world example: Assume a company has 10000 customer support 
+        agents and each agent spends on average 4 minutes with a 
+        customer.
+
+        >>> n = 10000
+        >>> z = np.random.default_rng().exponential(scale=4, size=n)
```
Comment (seberg) [inline numpy/random/_generator.pyx:390]: It seems you are calling it `d` below, so the test is failing.  But maybe best is to give it a meaningful name, one thought would be `call_durations = ...`?

However, I do wonder if an exponential is a good model for this (I would expect there to be a clear peak in typical call durations, which it does not have: there are no 2 second long calls), maybe we can use something that is known to follow an exponential distribution?

Comment (lzha97) [reply numpy/random/_generator.pyx:390]: @seberg Gotcha, I'll change the name to something more specific. What if I changed the problem context to be the number of minutes before a customer walks into a branch location for a store?

Comment (seberg) [reply numpy/random/_generator.pyx:390]: @lzha97, maybe the time between calls should be right, I think?  But I generally any poissant process should do (in physics radioactive decay, photons hitting a sensor, etc.)?  We could even just not add a close example.

Anyway, would be nice to push it forward, and the only thing is to fix the test and best make sure the example doesn't seem a strange fit with an exponential distribution!

**Comment by charris** [comment]
close/reopen

**Comment by charris** [comment]
Odd error here for the PR Labeler. @mattip I suspect the new permissions on general principles :) May need write permissions for the labeler.

**Comment by seberg** [comment]
@lzha97 for this PR, please make sure that the test:
```
azure-pipeline numpy.numpy (ComprehensiveTests macOS Python38)
```
passes.  You should be able to see the details if you click all the way through and ckeck the "Refguide Check".
The problem is currently:
```
File "build/testenv/lib/python3.8/site-packages/numpy/random/_generator.cpython-38-darwin.so", line ?, in Generator.exponential
Failed example:
    x-y
Expected:
    0.08 # will vary
Got:
    0.08330000000000004
```
Which seems OK, but I don't think the `will vary` is quite right.  IIRC it has to be `  # may vary`, importantly including the 2 spaces before the #.

**Comment by lzha97** [comment]
@seberg fixed the test and looks like all the checks have passed. Thanks so much for patiently helping me through it!

**Review by InessaPawson — APPROVED** [review]

**Comment by InessaPawson** [comment]
@seberg It's ready to be merged.

**Comment by melissawm** [comment]
Thank you @lzha97 !

**Pull request merged**